### ISSUE TITLE
Use logging functionality

### ISSUE THREAD

**Issue by matthiasdiener** [illinois-ceesd/timing#1]
https://github.com/anderson2981/CEESD-Y1_nozzle already has logging enabled. We should probably store the (aggregated) output sqlite file and maybe use this to measure the startup time, time step time, and other metrics.

```bash
# Gather output sqlite files from all ranks:
$ runalyzer-gather log.sqlite *.sqlite
# Add log.sqlite to this repo, maybe renaming the file to some time stamp

# Print execution time for time steps 10-20:
$ runalyzer -m log.sqlite -c 'print(sum(p[0] for p in q("select $t_step.max").fetchall()[10:20]))'

# Print max. startup time:
$ runalyzer -m log.sqlite -c 'print(q("select $t_init.max").fetchall()[0][0])'
```

**Comment by matthiasdiener** [comment]
(Side note: might have to install psutil manually to get startup time: `$ pip install psutil`)

**Comment by inducer** [comment]
I agree. That would certainly be nicer than my text/patch hack job.

**Comment by MTCam** [comment]
Should we save and update an sqlite file instead of a [text file](https://github.com/MTCam/timing/blob/lassen-auto-timing/y1-nozzle-timings.txt) for storing the testing data?

**Comment by lukeolson** [comment]
I'd prefer a text file here (or a text file for each run).

**Comment by matthiasdiener** [comment]
> Should we save and update an sqlite file instead of a [text file](https://github.com/MTCam/timing/blob/lassen-auto-timing/y1-nozzle-timings.txt) for storing the testing data?

Do you mean to save the results on each execution or just the timing data for plotting? For the results, I would store the raw testing data as a separate sqlite file for each run, in the way I outlined in the first message. This allows storing more data than a text file for later analysis. For the plotting, if you need to store a file in the repo at all, a text file is probably easier.

**Comment by inducer** [comment]
If we're using git/github as the place to store the data, then I think text is a better fit. Git's delta compression is shockingly efficient on text, but the same isn't necessary true for (binary) sqlite files.

**Comment by MTCam** [comment]
> 
> Do you mean to save the results on each execution or just the timing data for plotting? For the results, I would store the raw testing data as a separate sqlite file for each run, in the way I outlined in the first message. This allows storing more data than a text file for later analysis. For the plotting, if you need to store a file in the repo at all, a text file is probably easier.

What would be your proposed workflow for saving the sqlite files for every run?  Where do we put them? Archive them on each platform?

**Comment by matthiasdiener** [comment]
> > Do you mean to save the results on each execution or just the timing data for plotting? For the results, I would store the raw testing data as a separate sqlite file for each run, in the way I outlined in the first message. This allows storing more data than a text file for later analysis. For the plotting, if you need to store a file in the repo at all, a text file is probably easier.
> 
> What would be your proposed workflow for saving the sqlite files for every run? Where do we put them? Archive them on each platform?

I could imagine storing them in this repo. Seems like that would take about 100KB for each file.

**Comment by inducer** [comment]
> I could imagine storing them in this repo. Seems like that would take about 100KB for each file.

Not super opposed. That's 36 MB for a year, so not totally out of the question.

**Comment by MTCam** [comment]
> > I could imagine storing them in this repo. Seems like that would take about 100KB for each file.
> 
> Not super opposed. That's 36 MB for a year, so not totally out of the question.

I'd feel way better about this if we kept a single sqlite db file that is updated with the data from each run. We don't have 36MB of data from each run; not even close.  It just _feels_ wrong to duplicate that much overhead for each run.    

Also, that's 36MB *for each test* that we run, if we run each test only once daily.  I envision we 'll add more tests to this infrastructure, and potentially they could fire multiple times daily (e.g. with each commit that could impact performance).

**Comment by MTCam** [comment]
I think #2 closes this now. Perhaps the discussion regarding saving sqlite could be continued as another issue?
